### PR TITLE
treegrid group by load on demand sample - vNext

### LINKED ISSUE #2658: Create a Tree Grid group-by demo with data loaded on demand
### Description  
The idea is to demonstrate the functionality we have in the current group-by demo but have the data loaded on demand. 
Initially the grid should display the top-level groups only along with the summaries and reveal the inner ones on expansion.

### PR DESCRIPTION
**Pull request by igdmdimitrov**
Closes #2658

### CODE REVIEW

File: src/app/tree-grid/tree-grid-group-by-load-on-demand-sample/tree-grid-group-by-load-on-demand-sample.component.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            </igx-grid-toolbar-actions>
+        </igx-grid-toolbar>
+
+        <igx-column [field]="groupColumnKey" [width]="'180px'" [resizable]='true' [disableHiding]="true"></igx-column>
+
+        <igx-column field="OrderID" header="Order ID">
+        </igx-column>
+        <igx-column field="ShipCountry" header="Ship Country" width="200px" [groupable]="true">
+        </igx-column>
+        <igx-column field="OrderDate" header="Order Date" width="200px" [formatter]="formatDate">
```
Comment (DiyanDimitrov) [inline src/app/tree-grid/tree-grid-group-by-load-on-demand-sample/tree-grid-group-by-load-on-demand-sample.component.html:33]:
```suggestion
        <igx-column field="OrderDate" header="Order Date" width="200px" dataType="date">
```

File: src/app/tree-grid/tree-grid-group-by-load-on-demand-sample/tree-grid-group-by-load-on-demand-sample.component.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        </igx-column>
+        <igx-column field="ShipName" header="Ship Name" width="250px" [groupable]="true">
+        </igx-column>
+        <igx-column field="ShipCity" header="Ship City" width="250px" [groupable]="true">
+        </igx-column>
+        <igx-column field="ShipperName" header="Shipper Name" width="250px" [groupable]="true">
+        </igx-column>
+        <igx-column field="Salesperson" header="Salesperson" width="250px" [groupable]="true">
+        </igx-column>
+        <igx-column field="UnitPrice" header="Unit Price" width="150px" [formatter]="formatCurrency" dataType="number">
```
Comment (DiyanDimitrov) [inline src/app/tree-grid/tree-grid-group-by-load-on-demand-sample/tree-grid-group-by-load-on-demand-sample.component.html:47]:
```suggestion
        <igx-column field="UnitPrice" header="Unit Price" width="150px" dataType="currency">
```

File: src/app/tree-grid/tree-grid-group-by-load-on-demand-sample/tree-grid-group-by-load-on-demand-sample.component.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        this.reloadData();
+    }
+
+    public formatDate(val: Date) {
+        return new Intl.DateTimeFormat('en-US').format(val);
+    }
+
+    public formatCurrency(value: number) {
+        return value ? '$' + value.toFixed(3) : '';
+    }
```
Comment (DiyanDimitrov) [inline src/app/tree-grid/tree-grid-group-by-load-on-demand-sample/tree-grid-group-by-load-on-demand-sample.component.ts:52]: We could use the built-in column data types instead.
```suggestion
```

File: src/app/tree-grid/tree-grid-group-by-load-on-demand-sample/tree-grid-group-by-load-on-demand-sample.component.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    public ngAfterViewInit() {
+        this.groupColumns.forEach(col => {
+            const column = this.treeGrid.getColumnByName(col);
+            if (column) {
+                column.hidden = !column.hidden;
+            }
+        });
+        this.treeGrid.reflow();
```
Comment (DiyanDimitrov) [inline src/app/tree-grid/tree-grid-group-by-load-on-demand-sample/tree-grid-group-by-load-on-demand-sample.component.ts:34]: Calling `reflow` seems unnecessary.
```suggestion
```

**Review by DiyanDimitrov — CHANGES_REQUESTED** [review]

**Review by DiyanDimitrov — DISMISSED** [review]

**Comment by gedinakova** [comment]
@igdmdimitrov An NG0100 error is thrown in the console on loading the sample caused by the column hiding in ngAfterViewInit. I suggest we set the three columns' hidden property in the template and remove this code.

**Comment by gedinakova** [comment]
@igdmdimitrov On adding or removing a column from the group-by area, the grid's header is not refreshed immediately and the column is present/not present until expansion.

**Review by gedinakova — APPROVED** [review]

**Review by DiyanDimitrov — APPROVED** [review]

**Pull request merged**